llvm/lib/CheerpUtils/PointerAnalyzer.cpp

```diff
@@ -1468,6 +1468,12 @@ void PointerAnalyzer::prefetchFunc(const Function& F) const
 		IndirectPointerKindConstraint returnConstraint(RETURN_CONSTRAINT, &F);
 		PACache.pointerKindData.constraintsMap[returnConstraint] = RAW;
 	}
+	else if (TypeSupport::isJSExportedPtrType(retType, *F.getParent()))
+	{
+		IndirectPointerKindConstraint returnConstraint(RETURN_CONSTRAINT, &F);
+		PACache.pointerKindData.constraintsMap[returnConstraint] = PointerKindWrapper(SPLIT_REGULAR, &F);
```

Comment:
There should be an independently tracked set of constraints just for jsexport, which correponds to the requirements for any argument of that type across all jsexported methods